test/jdk/java/foreign/TestStringEncoding.java

```diff
@@ -250,6 +250,32 @@ public void segmentationFault() {
         }
     }
 
+    // This test ensures that we do not address outside the segment even though there
+    // are odd bytes at the end.
+    @Test(dataProvider = "strings")
+    public void offBoundaryTrailingBytes(String testString) {
+        if (testString.length() < 3 || !containsOnlyRegularCharacters(testString)) {
+            return;
+        }
+        for (var charset : standardCharsets()) {
+            for (var arena: arenas()) {
+                try (arena) {
+                    MemorySegment strSegment = arena.allocateFrom(testString, charset);
```

Comment:
This could just use String::getBytes ? E.g. you don't really need the `strSegment`